scipy/interpolate/_aaa.py

```diff
@@ -198,6 +199,10 @@ def __init__(self, points, values, *, rtol=None, max_terms=100):
         
         if not np.all(np.isfinite(z)):
             raise ValueError("`points` must be finite.")
+        
+        if max_terms < 1 or not np.issubdtype(type(max_terms), np.integer):
+            raise ValueError("`max_terms` must be an integer value greater than or "
```

Comment:
Nit : am not a fan of subdtype(np.integer). This is what operator.index is for

Reply:
Happy to change, suggesting ditching the  `not np.issubdtype(type(max_terms), np.integer)`?

Reply:
exactly. 
```
max_terms = operator.index(max_terms)    # raises if not integer
if max_terms < 0:
   raise ValueError(....)
```